ts/src/main/scala/ch/epfl/ts/traders/MadTrader.scala

```diff
@@ -0,0 +1,100 @@
+package ch.epfl.ts.traders
+
+import scala.util.Random
+import scala.concurrent.duration.FiniteDuration
+import ch.epfl.ts.data.LimitAskOrder
+import ch.epfl.ts.data.LimitBidOrder
+import ch.epfl.ts.data.Order
+import ch.epfl.ts.component.StartSignal
+import ch.epfl.ts.data.Currency
+import ch.epfl.ts.data.StrategyParameters
+import ch.epfl.ts.data.ParameterTrait
+import ch.epfl.ts.data.CurrencyPairParameter
+import ch.epfl.ts.data.MarketAskOrder
+import ch.epfl.ts.data.MarketBidOrder
+import ch.epfl.ts.data.CoefficientParameter
+import ch.epfl.ts.data.NaturalNumberParameter
+import ch.epfl.ts.data.TimeParameter
+import ch.epfl.ts.component.ComponentBuilder
+
+/**
+ * Required and optional parameters used by this strategy
+ */
+object MadTrader extends TraderCompanion {
+  type ConcreteTrader = MadTrader
+  override protected val concreteTraderTag = scala.reflect.classTag[MadTrader]
+  
+  /** Interval between two random trades (in ms) */
+	val INTERVAL = "interval"
+
+	/** Initial delay before the first random trade (in ms) */
+  val INITIAL_DELAY = "initial_delay"
+
+  /** Volume of currency to trade (in currency unit) */
+  val ORDER_VOLUME = "order_volume"
+  /** Random variations on the volume (in percentage of the order volume) */
+  val ORDER_VOLUME_VARIATION = "order_volume_variation"
+
+  /** Which currencies to trade */
+  val CURRENCY_PAIR = "currency_pair"
+
+  override def requiredParameters: Map[Key, ParameterTrait] = Map(
+      INTERVAL -> TimeParameter,
+      ORDER_VOLUME -> NaturalNumberParameter,
+      CURRENCY_PAIR -> CurrencyPairParameter
+    )
+  override def optionalParameters: Map[Key, ParameterTrait] = Map(
+      INITIAL_DELAY -> TimeParameter,
+      ORDER_VOLUME_VARIATION -> CoefficientParameter
```

Comment:
It would actually be nice to have both positive and negative variation here, but it's a dummy example anyway :tongue: